modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/LongRunningQueryTest.java

```diff
@@ -182,6 +220,106 @@ public void testLongLocalLazy() {
         checkFastQueries();
     }
 
+    /** */
+    @Test
+    public void testLongRunningInsert() {
+        local = false;
+
+        runDml(INSERT, false);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningInsertWithSubquery() {
+        local = false;
+
+        runDml(INSERT_WITH_SUBQUERY, true);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningInsertLocal() {
+        local = true;
+
+        runDml(INSERT, false);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningInsertWithSubqueryLocal() {
+        local = true;
+
+        runDml(INSERT_WITH_SUBQUERY, true);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningUpdate() {
+        local = false;
+
+        runDml(UPDATE, false);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningUpdateWithSubquery() {
+        local = false;
+
+        runDml(UPDATE_WITH_SUBQUERY, true);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningUpdateLocal() {
+        local = true;
+
+        runDml(UPDATE, false);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningUpdateWithSubqueryLocal() {
+        local = true;
+
+        runDml(UPDATE_WITH_SUBQUERY, true);
+    }
+
+    /** */
+    @Test
+    public void testLongRunningDelete() {
+        local = false;
+
+        increaseCacheSize(1_000_000);
```

Comment:
Test can become flaky if delete of such amount will be executed too fast. Let's use the same approach as for update (for example "WHERE sleep_func(?) > 0"

Reply:
The delete statement is corrected.